SwiftCompilerSources/Sources/SIL/ForwardingInstruction.swift

```diff
@@ -397,3 +397,33 @@ extension LinearFunctionExtractInst: ForwardingInstruction {
   public var canForwardGuaranteedValues: Bool { true }
   public var canForwardOwnedValues: Bool { true }
 }
+
+// -----------------------------------------------------------------------------
+// ownership transition instructions
+
+/// An instruction that transfers lifetime dependence from a single operand to a single result. The operand and result
+/// have the same identity, but they are not part of the same forwarded lifetime:
+/// copy_value, move_value, begin_borrow.
+public protocol OwnershipTransitionInstruction: UnaryInstruction {
```

Comment:
nit: These are in "ForwardindInstruction".swift. Is there any other more appropriate file to put them ?

Reply:
No strong opinion. But I think it's important to understand the relationship between ForwardingInstruction and OwnershipTransitionInstruction. It's easy to get them confused and an instruction is either one or the other but never both